code/crates/core-driver/src/mux.rs

```diff
@@ -216,6 +216,41 @@ where
         None
     }
 
+    pub(crate) fn store_and_multiplex_polka_certificate(
```

Comment:
Ref: https://github.com/informalsystems/malachite/blob/ad7350e81006dd28a1351c4eb33a2349208951cf/specs/consensus/quint/votekeeper.qnt#L445

Reply:
Looking at the spec again, I believe this method is not correct. Instead of sending a `ProposalAndPolkaCurrent` to the state machine, we should instead follow the same logic as the spec and override the votes in the vote keeper.

We should add the `apply_certificate` method to the `VoteKeeper` struct and implement that logic there.

Reply:
We can do that. But we can also store/ retrieve the certificate as you do here. In this case we also have to change `multiplex_proposal()` and how we determine `polka_current` and `polka_previous`.